src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,233 @@
+}
+
+func doStartCommand(cmd *cobra.Command) error {
+	var commandLineArgs []string
+
+	if cmd.Flags().Changed("container-engine") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-container-engine", specifiedSettings.ContainerEngine)
+	}
+	if cmd.Flags().Changed("kubernetes-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-enabled", strconv.FormatBool(specifiedSettings.Enabled))
+	}
+	if cmd.Flags().Changed("kubernetes-version") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-version", specifiedSettings.Version)
+	}
+	if cmd.Flags().Changed("flannel-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-options-flannel", strconv.FormatBool(specifiedSettings.Flannel))
+	}
+	if applicationPath == "" {
+		pathLookupFuncs := map[string]func(rdctlPath string) string{
+			"windows": getWindowsRDPath,
+			"linux":   getLinuxRDPath,
+			"darwin":  getMacOSRDPath,
+		}
+		getPathFunc, ok := pathLookupFuncs[runtime.GOOS]
+		if !ok {
+			return fmt.Errorf("don't know how to find the path to Rancher Desktop on OS %s", runtime.GOOS)
+		}
+		rdctlPath, err := os.Executable()
+		if err != nil {
+			rdctlPath = ""
+		}
+		applicationPath = getPathFunc(rdctlPath)
+		if applicationPath == "" {
+			return fmt.Errorf("no executable for Rancher Desktop found in the default locations; please retry with the --path|-p option")
+		}
+	}
+	return launchApp(applicationPath, commandLineArgs)
+}
+
+func launchApp(applicationPath string, commandLineArgs []string) error {
+	var commandName string
+	var args []string
+
+	if runtime.GOOS == "darwin" {
+		commandName = "/usr/bin/open"
+		args = []string{"-a", applicationPath}
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func moveToParent(fullPath string, numberTimes int) string {
+	fullPath = path.Clean(fullPath)
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+func getWindowsRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	localAppDataGetters := []func() string{
+		func() string {
+			return os.Getenv("LOCALAPPDATA")
+		},
+		func() string {
+			homeDrive := os.Getenv("HOMEDRIVE")
+			if homeDrive == "" {
+				return ""
+			}
+			homePathPart := os.Getenv("HOMEPATH")
+			if homePathPart == "" {
+				return ""
+			}
```

Comment:
Sure. It probably doesn't matter because we still need to find an executable called "Rancher Desktop.exe" under at least one of those directories, but easy to change.